src/js/components/Layer/Layer.js

```diff
@@ -1,76 +1,78 @@
-        }, 0);
-      } else if (
-        this.originalFocusedElement.parentNode &&
-        this.originalFocusedElement.parentNode.focus
-      ) {
-        // required for IE11 and Edge
-        this.originalFocusedElement.parentNode.focus();
+    return () => {
+      const originalFocusedElement = originalFocusedElementRef.current;
+
+      if (originalFocusedElement) {
+        if (originalFocusedElement.focus) {
+          // wait for the fixed positioning to come back to normal
+          // see layer styling for reference
+          setTimeout(() => {
+            originalFocusedElement.focus();
+          }, 0);
+        } else if (
+          originalFocusedElement.parentNode &&
+          originalFocusedElement.parentNode.focus
+        ) {
+          // required for IE11 and Edge
+          originalFocusedElement.parentNode.focus();
+        }
       }
-    }
 
-    const activeAnimation = animation !== undefined ? animation : animate;
-    if (activeAnimation !== false) {
-      // undefined uses 'slide' as the default
-      // animate out and remove later
-      const layerClone = this.layerContainer.cloneNode(true);
-      layerClone.id = 'layerClone';
-      document.body.appendChild(layerClone);
-      const clonedContainer = layerClone.querySelector(
-        '[class*="StyledLayer__StyledContainer"]',
-      );
-      clonedContainer.style.animationDirection = 'reverse';
-      setTimeout(() => {
-        // we add the id and query here so the unit tests work
-        const clone = document.getElementById('layerClone');
-        if (clone) document.body.removeChild(clone);
-      }, animationDuration);
-    }
-  }
+      const activeAnimation = animation !== undefined ? animation : animate;
+      if (activeAnimation !== false) {
+        // undefined uses 'slide' as the default
+        // animate out and remove later
+        const layerClone = layerContainerRef.current.cloneNode(true);
+        layerClone.id = 'layerClone';
+        document.body.appendChild(layerClone);
+        const clonedContainer = layerClone.querySelector(
+          '[class*="StyledLayer__StyledContainer"]',
+        );
+        clonedContainer.style.animationDirection = 'reverse';
+        setTimeout(() => {
+          // we add the id and query here so the unit tests work
+          const clone = document.getElementById('layerClone');
+          if (clone) document.body.removeChild(clone);
+        }, animationDuration);
+      }
+    };
+  }, [animation, animate]);
 
-  render() {
-    const { islayerContainerAvailable } = this.state;
+  return isLayerContainerAvailable
+    ? createPortal(<LayerContainer {...props} />, layerContainerRef.current)
+    : null;
+};
 
-    return islayerContainerAvailable
-      ? createPortal(<LayerContainer {...this.props} />, this.layerContainer)
-      : null;
-  }
-}
+Layer.defaultProps = {
+  full: false,
+  margin: 'none',
+  modal: true,
+  position: 'center',
+  responsive: true,
+};
```

Comment:
Can we please move the default props assignment to be part of the functional component declaration (line #9)?

Reply:
If we define all the `defaultProps` on line #9, in the function parameters, then we would also need to pass every single prop into the `LayerContainer` component manually. (line #65)
That feels a bit awkward to me.
This is the only reason I didn't destructure the two props needed on line #9 but did it on line #10 so we could just pass along the props as-is to the `LayerContainer` component.

If being explicit is a requirement, I have no problem doing it the way you recommend. I just expressed my reasoning.